sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -11,13 +12,19 @@
 static const az_span provisioning_service_api_version
     = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
 
-static const az_span str_dps = AZ_SPAN_LITERAL_FROM_STR("$dps");
-static const az_span str_registrations = AZ_SPAN_LITERAL_FROM_STR("/registrations/");
+static const az_span user_agent_version_prefix = AZ_SPAN_LITERAL_FROM_STR("&ClientVersion=");
+
+static const az_span str_dps_registrations_res
+    = AZ_SPAN_LITERAL_FROM_STR("$dps/registrations/res/");
+static const int32_t idx_registrations_start = 4;
+static const int32_t idx_registrations_end = 19;
+
+static const int32_t idx_res_start = 19;
```

Comment:
registration end and res start are the same number? Do we want one before the `/` and one after, or is this just to make the calling code more readable.

Reply:
Right: this is for readability and to avoid to declare global spans that would use more flash. (Maybe I'm over-optimizing here.)

Reply:
I think global spans will help readability/maintainability more, but your call :)

Reply:
You were right about idx_res_start - removing that one.
Global spans are larger though right (ptr, size) and this is just the index.

Reply:
> Global spans are larger though right (ptr, size) and this is just the index.

Yep, this is just trade offs around code readability/self documenting code vs. perf/size. You know the constraints well so I will defer to your judgement. An index + comment is likely a good enough compromise :)

Reply:
I think I have a solution to meet both requirements: I'll encapsulate the indexes within two inline functions that will return the span.